docs/algorithms/Legendre's Algorithm/legendre-algo.md

```diff
@@ -0,0 +1,262 @@
+---
+id: legendre-algo
+title: Finding Power of Factorial Divisor
+sidebar_label: Legendre's Formula
+---
+
+# Deep Dive into Finding the Power of a Factorial Divisor
+
+
+This algorithm is used to find the largest power x such that k^x divides n! , where n! is the factorial of a given number n , and k can be either a prime or composite number.
+
+
+## Why This Algorithm is Useful
+
+
+In number theory and combinatorics, determining the largest power of a divisor for factorials is crucial in problems involving:
+
+
+- **Divisibility in large products** (e.g., does n! divide a large power of some integer?)
+- **Prime factorization in combinatorial coefficients** (e.g., the power of a prime factor in binomial coefficients)
+- **Factorial simplifications in modular arithmetic** (useful in combinatorics, cryptography, etc.)
+
+
+This algorithm is especially valuable because calculating n! directly for large n is computationally infeasible, but by using Legendre's formula, we can determine the power of a divisor without computing the factorial itself.
+
+
+---
+
+
+## Understanding the Problem for Prime k 
+
+
+Let's first consider the case where k is a **prime number**.
+
+
+### Example Problem: Power of Prime Factor Dividing n! 
+
+
+Suppose we want to find the largest power of a prime k = 2 that divides n! , where n = 10:
+
+
+$$ 10! = 1 * 2 * 3 * . . . * 10 = 3628800 $$
+
+
+
+We need to find the highest power x such that 2^x divides 3628800.
+
+
+### Derivation of Legendre's Formula
+
+
+1. **Counting Divisors of Prime k :** 
+
+
+   The factorial $$ n! = 1 * 2 * 3 * n $$ contains various multiples of k, but not every number from 1 to n is divisible by k.
+
+
+2. **Step-by-Step Divisor Counting for Powers of k :**
+
+
+   - **First Power k :** Every k-th number (i.e., every multiple of k) in the product from 1 to n contributes a factor of k . There are $$\left\lfloor \frac{n}{k} \right\rfloor$$ such numbers.
```

Comment:
replace `$$\left\lfloor \frac{n}{k} \right\rfloor$$` with `$\left\lfloor \frac{n}{k} \right\rfloor$` because `$` for inline and `$$` for block line elements.